client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -986,7 +988,20 @@ private void requireNonBatchStatement()
         }
     }
 
-    private static String getExecuteSql(String statementName, List<String> values)
+    private String getExecuteImmediateSql(List<String> values)
+    {
+        StringBuilder sql = new StringBuilder();
+        sql.append("EXECUTE IMMEDIATE '");
+        sql.append(this.originalSql.replaceAll("'", "''"));
+        sql.append("'");
```

Comment:
@electrum Is there some class which already has the formatting logic for prepared statements already?

i.e. is the `'` -> `''` the only transformation which needs to be done?

Reply:
The statement needs to be a proper string literal.  We do have a `formatStringLiteral()` method in this class (used for formatting parameters).  We could use that (it doesn't have additional functionality).

Reply:
Updated to use `formatStringLiteral()`